org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java

```diff
@@ -388,6 +381,16 @@ public void visitEnd() {
 			if (r != m) {
 				i.merge(nodeToInstruction.get(r));
 			}
+
+			if (!ignored.contains(m)) {
+				final int line = i.getLine();
+				if (firstLine > line || lastLine == ISourceNode.UNKNOWN_LINE) {
+					firstLine = line;
```

Comment:
@marchof IMO this is good change on its own - can extract into separate PR if desired.